importer/RuRu.py

```diff
@@ -0,0 +1,107 @@
+from Monument import Monument, Dataset
+import importer_utils as utils
+import importer as importer
+
+
+class RuRu(Monument):
+
+    def set_adm_location(self):
+        """
+        Set the administrative location.
+
+        Preferably try to resolve the 'district' (район/округ,
+        the field contains different types so that's why we're
+        matching it against the whole big list of administrative
+        divisions of different kinds).
+        If that doesn't work, use the higher level субъект.
+        """
+        level_three = self.data_files["administrative"]
+        level_two = self.data_files["subyekty"]
+        district_match = [x for x in level_three if x[
+            "itemLabel"] == self.district]
+        if len(district_match) == 1:
+            self.add_statement("located_adm", district_match[0]["item"])
+        else:
+            sub_match = [x for x in level_two if x[
+                "value"].lower() == self.region_iso]
+            if len(sub_match) == 1:
+                self.add_statement("located_adm", sub_match[0]["item"])
+            else:
+                self.add_to_report("district", self.district, "located_adm")
+
+    def set_location(self):
+        """
+        Set the location property using 'city' field.
+
+        Of course it's often a village etc. :)
+        Make sure that the target page has a P31 of a
+        settlement, to avoid disambigs etc.
+        Also, compare with 'administrative location' of the
+        object, and don't add if it's the same.
+        """
+        if self.has_non_empty_attribute("city"):
+            city_q_try = utils.q_from_wikipedia("ru", self.city)
+            try:
+                administrative_q = self.wd_item[
+                    "statements"]["P131"][0]["value"]
+            except KeyError:
+                administrative_q = None
+            if city_q_try and city_q_try != administrative_q:
+                city_q_is = utils.get_P31(city_q_try, self.repo)
+                for p31_value in city_q_is:
+                    if p31_value in self.data_files["settlement"]:
+                        self.add_statement("location", city_q_try)
+                        # there can be more than one P31,
+                        # but after first positive
+                        # we can leave
+                        return
+                    else:
```

Comment:
I think we want the report on the same level as the loop. Otherwise we will get a report for the first 4 non-settlement P31s even if the fifth is a settlement.